src/libcoretest/ptr.rs

```diff
@@ -89,6 +101,15 @@ fn test_as_ref() {
             let p: *const int = &u as *const _;
             assert_eq!(p.as_ref().unwrap(), &2);
         }
+
+        let sn = slice::from_raw_parts(null(), 0) as *const [u32];
+        assert_eq!(sn.as_ref(), None);
```

Comment:
This is "6.1.3.2.3 Behavior considered undefined" because we put a null pointer inside a `&[T]`. It's just a test case, so I guess it can do no harm. I don't know what I want to say more than this is UB and rustc doesn't have to compile it into a sane test case if it doesn't want to.